fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -2318,6 +2323,18 @@ public void mergeIsomorphicProfiles() {
             List<RuntimeProfile> instanceProfiles = fragmentProfile.getChildList().stream()
                     .map(pair -> pair.first)
                     .collect(Collectors.toList());
+
+            // Setup backend address infos
+            Set<String> backendAddresses = Sets.newHashSet();
+            instanceProfiles.forEach(instanceProfile -> {
+                Matcher matcher = ADDRESS_PATTERN.matcher(instanceProfile.getName());
```

Comment:
do we must use regex？ regex is cpu intensive

Reply:
I have condunct a benchmark, and it shows that regex is two times slower than the non-regex way, but the time usage is microsecond level, so I think it is acceptable.

```
Benchmark            Mode  Cnt     Score     Error  Units
Regex.parseByNormal  avgt    5   640.062 ± 152.269  ns/op
Regex.parseByRegex   avgt    5  1286.652 ±  68.358  ns/op
```


```java
package org.liuyehcf.jmh.serialize;

import org.openjdk.jmh.annotations.*;
import org.openjdk.jmh.runner.Runner;
import org.openjdk.jmh.runner.options.Options;
import org.openjdk.jmh.runner.options.OptionsBuilder;

import java.util.concurrent.TimeUnit;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

@Warmup(iterations = 1, time = 1, timeUnit = TimeUnit.SECONDS)
@Measurement(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
@Fork(1)
@Threads(1)
@BenchmarkMode(Mode.AverageTime)
@OutputTimeUnit(TimeUnit.NANOSECONDS)
public class Regex {
    private static final Pattern ADDRESS_PATTERN =
            Pattern.compile(".*?TNetworkAddress\\(hostname:(.*?), port:([0-9]+?)\\).*?");

    private static final String INFO = "f33dd098-374c-11ed-aefc-5254007b45e8 (host=TNetworkAddress(hostname:172.26.92.195, port:16670)):(Active: 95.230ms[95230578ns], % non-child: 0.00%)";
    private static final String EXPECTED_RESULT = "172.26.92.195:16670";

    public static void main(String[] args) throws Exception {
        Options opt = new OptionsBuilder()
                .include(Regex.class.getSimpleName())
                .build();

        new Runner(opt).run();
    }

    @Benchmark
    public void parseByRegex() {
        Matcher matcher = ADDRESS_PATTERN.matcher(INFO);
        if (matcher.matches()) {
            String address = String.format("%s:%s", matcher.group(1), matcher.group(2));
            if (!EXPECTED_RESULT.equals(address)) {
                throw new RuntimeException();
            }
        }
    }

    @Benchmark
    public void parseByNormal() {
        String prefix = "host=TNetworkAddress(";
        String suffix = ")):(Active:";
        int startIndex = INFO.indexOf(prefix) + "host=TNetworkAddress(".length();
        int endIndex = INFO.indexOf(suffix);
        String[] segments = INFO.substring(startIndex, endIndex).split(", ");
        String host = segments[0].split(":")[1];
        String port = segments[1].split(":")[1];
        String address = String.format("%s:%s", host, port);
        if (!EXPECTED_RESULT.equals(address)) {
            throw new RuntimeException();
        }
    }
}
```